### ISSUE TITLE
Service Doctrine\DBAL\Driver\Connection is an instance of Doctrine\DBAL\Connection which does not implement Doctrine\DBAL\Driver\Connection

### ISSUE THREAD

**Issue by chrif** [doctrine/DoctrineBundle#1637]
"doctrine/dbal": "^3.6",
"doctrine/doctrine-bundle": "^2.8",

In dbal.xml:
`<service id="Doctrine\DBAL\Driver\Connection" alias="database_connection" public="false" />`

This service enables autowiring on the `Doctrine\DBAL\Driver\Connection` interface but `database_connection` is an instance of `Doctrine\DBAL\Connection` which does not implement `Doctrine\DBAL\Driver\Connection` anymore in DBAL 3.x

So why keeping this definition? Is it only for backward compatibility? But if it makes autowiring fails it's not really backward compatible.

**Comment by ostrolucky** [comment]
Right below that there is a same alias for `Doctrine\DBAL\Connection` which should explain why nobody complained till now. Or do you have some scenario where you get autowiring error regarding this?

**Comment by chrif** [comment]
I'm upgrading an app from dbal 2 to 3. It had a custom service like this:
```
  doctrine.connection:
    class: Doctrine\DBAL\Driver\Connection
    factory: ['@doctrine.orm.entity_manager', getConnection]
    public: true
```
and developers were in the habit of injecting the `Doctrine\DBAL\Driver\Connection` interface with autowiring.

I replaced all usages of `Doctrine\DBAL\Driver\Connection` to `Doctrine\DBAL\Connection` and replaced the service with this definition:
```
  doctrine.connection:
    alias: 'doctrine.dbal.default_connection'
    public: true
```

Now developers need to lose the habit of autowiring `Doctrine\DBAL\Driver\Connection` which I thought would fail with a "no such service error". I was surprised to see instead an "interface not implemented" error, which is confusing.

I created a compiler pass to remove it.
